overlord/snapshotstate/snapshotstate.go

```diff
@@ -385,6 +385,19 @@ func Save(st *state.State, instanceNames []string, users []string) (setID uint64
 		if err != nil {
 			return 0, nil, nil, err
 		}
+	} else {
+		var snapst snapstate.SnapState
+
+		for _, name := range instanceNames {
+			err := snapstate.Get(st, name, &snapst)
+			if err != nil && err != state.ErrNoState {
+				return 0, nil, nil, err
+			}
+
+			if !snapst.IsInstalled() {
```

Comment:
Nitpick: that's fine, but `state.ErrNoState` would do as well.

Reply:
We do have an `isInstalled()` function in `overlord/snapstate/handlers.go` and an `All()` function in `overlord/snapstate/snapstate.go` which returns a map.

I'm not sure if they would help; your call. :-)

Reply:
@MiguelPires @mardy it's actually a good idea indeed to use `All()` instead since `Get` needs to get all snaps every time which is wasteful.